File: src/libNet/netMsg/FailureMsg.cxx
/*       F A I L U R E . C X X
 * BRL-CAD
 *
 * Copyright (c) 2011 United States Government as represented by
 * the U.S. Army Research Laboratory.
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public License
 * version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this file; see the file named COPYING for more
 * information.
 */
/** @file FailureMsg.cxx
 *
 * Brief description
 *
 */

#include "FailureMsg.h"
#include "NetMsgTypes.h"
#include <sstream>

/* Normal Constructor */
FailureMsg::FailureMsg(uint8_t failureCode):
  GenericOneByteMsg(FAILURE, failureCode)
{}

/* Reply Constructor */
FailureMsg::FailureMsg(NetMsg* msg, uint8_t failureCode) :
  GenericOneByteMsg(FAILURE, msg, failureCode)
{}

/* Deserializing Constructor */
FailureMsg::FailureMsg(ByteBuffer* bb, Portal* origin):
  GenericOneByteMsg(bb, origin)
{}

/* Destructor */
FailureMsg::~FailureMsg()
{}
/*
 *Getters n Setters
 */
uint8_t FailureMsg::getFailureCode() {return this->data;}

/*
 * Local Variables:
 * mode: C
 * tab-width: 8
 * indent-tabs-mode: t
 * c-file-style: "stroustrup"
 * End:
 * ex: shiftwidth=4 tabstop=8
 */

File: src/interfaces/java/src/org/brlcad/geometryservice/net/msg/FailureMsg.java
/*
 * BRL-CAD
 *
 * Copyright (c) 2011 United States Government as represented by
 * the U.S. Army Research Laboratory.
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public License
 * version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this file; see the file named COPYING for more
 * information.
 */
/** @file FailureMsg.java
 * 
 */
package org.brlcad.geometryservice.net.msg;

import org.brlcad.geometryservice.net.ByteBufferReader;
import org.brlcad.geometryservice.net.ByteBufferWriter;

public class FailureMsg extends AbstractNetMsg {
	private byte failureCode;

	/**
	 * @param msgType
	 */
	public FailureMsg(byte failureCode) {
		super(NetMsgTypes.Failure);
		this.failureCode = failureCode;
	}

	/**
	 * @param msgType
	 * @param reMsg
	 */
	public FailureMsg(AbstractNetMsg reMsg, byte failureCode) {
		super(NetMsgTypes.Failure, reMsg);
		this.failureCode = failureCode;
	}

	/**
	 * @param msgType
	 * @param reader
	 */
	public FailureMsg(ByteBufferReader reader) {
		super(NetMsgTypes.Failure, reader);
	}

	/*
	 * (non-Javadoc)
	 * 
	 * @see
	 * org.brlcad.geometryservice.net.msg.AbstractNetMsg#_deserialize(org.brlcad
	 * .geometryservice.net.ByteBufferReader)
	 */
	@Override
	protected void _deserialize(ByteBufferReader reader) {
		this.failureCode = reader.get();
	}

	/*
	 * (non-Javadoc)
	 * 
	 * @see
	 * org.brlcad.geometryservice.net.msg.AbstractNetMsg#_serialize(org.brlcad
	 * .geometryservice.net.ByteBufferWriter)
	 */
	@Override
	protected void _serialize(ByteBufferWriter writer) {
		writer.put(this.failureCode);
	}

	public byte getFailureCode() {
		return failureCode;
	}

}
